packages/compass-shell/src/modules/worker-runtime/runtime.js

```diff
@@ -0,0 +1,106 @@
+import path from 'path';
+import fs from 'fs';
+// eslint-disable-next-line camelcase
+import child_process from 'child_process';
+import crypto from 'crypto';
+import electron from 'electron';
+import worker from 'inline-entry-loader!./worker';
+import { createCaller, exposeAll } from './rpc';
+
+function createChildFromSource(src) {
+  const tmpFile = path.resolve(
+    (electron.app || electron.remote.app).getPath('temp'),
+    `worker-runtime-${crypto.randomBytes(32).toString('hex')}`
+  );
+
+  // eslint-disable-next-line no-sync
+  fs.writeFileSync(tmpFile, src, 'utf8');
+
+  const childProcess = child_process.spawn(process.execPath, [tmpFile], {
+    stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
+    env: { ...process.env, ELECTRON_RUN_AS_NODE: 1 },
+    /**
+     * This would be awesome, but only supported since node 12.16.0, compass
+     * is currently at 12.4
+     * @see https://nodejs.org/api/child_process.html#child_process_advanced_serialization
+     */
+    serialization: 'advanced',
```

Comment:
You can emulate this quite closely by using [`v8.serialize()` and `v8.deserialize()`](https://nodejs.org/api/v8.html#v8_serialization_api) – they return Buffers, so you would have to perform that + base64 or similar in order to use process.send, but it might be worth it? (Also, thank you for saying that this feature is awesome :wink:)

Reply:
Omg, Anna, you worked on so much great features in Node! 😻

Reply:
Works like a charm! 🎉 

Added in c68093f5577a66da8b5bda16e7557d2b034c9c6b